### ISSUE TITLE
Feature Tours trap focus and blurs focus after clearing (Accessibility)

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#6641]
## Bug Description

The feature tours steal focus after dismissing a notification, which is a bit odd. Worse is the feature tour trapping focus, but combined with it then blurring user focus and returning you to the first element of the screen, the experience is confusing.

## Steps to reproduce

1. Turn on VoiceOver and trigger any feature tour
2. Try to tab through content—notice focus is trapped inside the feature tour content
3. Exit the feature tour with the the "close" button, "Got it" button, or "Escape" key (should work, but actually haven't tested it—if it doesn't, this should work too)
4. Notice focus is not restored to the previous location but is blurred and no element is focused. User must tab to restore visual focus to an element on the page.

## Screenshots

https://user-images.githubusercontent.com/90871/221290644-bcf1a6c1-ee70-4b17-b3c7-e77036102ea1.mp4

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When a feature tour is displayed and focus is stolen from keyboard user, the previously-selected element should be stored.
* When the feature tour is dismissed, focus should be restored to the previous location.
  * If this is not possible in a predictable way, keyboard focus should be restored to the Site Kit Dashboard, somewhere in the menu at the top of the dashboard is acceptable.

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by techanvil** [comment]
Update: I've paused on this issue pending a reply on this [question](https://github.com/google/site-kit-wp/issues/6638#issuecomment-1620470774) for @tofumatt on #6638, as depending on the outcome it's possible this issue may no longer be needed.

**Comment by tofumatt** [comment]
@techanvil Answered in that issue, but you're right: we can fix both issues from #6638, so closing this one in favour of that one. 👍🏻
